internal/trie/tracking/deltas_test.go

```diff
@@ -0,0 +1,182 @@
+// Copyright 2022 ChainSafe Systems (ON)
+// SPDX-License-Identifier: LGPL-3.0-only
+
+package tracking
+
+import (
+	"testing"
+
+	"github.com/ChainSafe/gossamer/lib/common"
+	"github.com/stretchr/testify/assert"
+)
+
+func Test_New(t *testing.T) {
+	t.Parallel()
+
+	deltas := New()
+
+	expectedDeltas := &Deltas{
+		deletedNodeHashes: make(map[common.Hash]struct{}),
+	}
+	assert.Equal(t, expectedDeltas, deltas)
+}
+
+func Test_Deltas_RecordDeleted(t *testing.T) {
+	t.Parallel()
+
+	testCases := map[string]struct {
+		deltas         Deltas
+		nodeHash       common.Hash
+		expectedDeltas Deltas
+	}{
+		"set in empty deltas": {
+			deltas: Deltas{
+				deletedNodeHashes: map[common.Hash]struct{}{},
+			},
+			nodeHash: common.Hash{1},
+			expectedDeltas: Deltas{
+				deletedNodeHashes: map[common.Hash]struct{}{{1}: {}},
+			},
+		},
+		"set in non empty deltas": {
+			deltas: Deltas{
+				deletedNodeHashes: map[common.Hash]struct{}{{1}: {}},
+			},
+			nodeHash: common.Hash{2},
+			expectedDeltas: Deltas{
+				deletedNodeHashes: map[common.Hash]struct{}{
+					{1}: {}, {2}: {},
+				},
+			},
+		},
+		"override in deltas": {
+			deltas: Deltas{
+				deletedNodeHashes: map[common.Hash]struct{}{{1}: {}},
+			},
+			nodeHash: common.Hash{1},
+			expectedDeltas: Deltas{
+				deletedNodeHashes: map[common.Hash]struct{}{{1}: {}},
+			},
+		},
+	}
+
+	for name, testCase := range testCases {
+		testCase := testCase
+		t.Run(name, func(t *testing.T) {
+			t.Parallel()
+
+			testCase.deltas.RecordDeleted(testCase.nodeHash)
+			assert.Equal(t, testCase.expectedDeltas, testCase.deltas)
+		})
+	}
+}
+
+func Test_Deltas_Deleted(t *testing.T) {
+	t.Parallel()
+
+	testCases := map[string]struct {
+		deltas     Deltas
+		nodeHashes map[common.Hash]struct{}
+	}{
+		"empty deltas": {},
+		"non empty deltas": {
+			deltas: Deltas{
+				deletedNodeHashes: map[common.Hash]struct{}{{1}: {}},
+			},
+			nodeHashes: map[common.Hash]struct{}{{1}: {}},
+		},
+	}
+
+	for name, testCase := range testCases {
+		testCase := testCase
+		t.Run(name, func(t *testing.T) {
+			t.Parallel()
+
+			nodeHashes := testCase.deltas.Deleted()
+			assert.Equal(t, testCase.nodeHashes, nodeHashes)
+		})
+	}
+}
+
+func Test_Deltas_MergeWith(t *testing.T) {
+	t.Parallel()
+
+	testCases := map[string]struct {
+		deltas         Deltas
+		deltasArg      DeletedGetter
+		expectedDeltas Deltas
+	}{
+		"merge empty deltas": {
```

Comment:
could you change the names to `snake_case` pattern?

Reply:
Done 👍